modules/atomic-widgets/dynamic-tags.php

```diff
@@ -167,19 +180,46 @@ private function convert_text_control_to_atomic( $control ) {
 		];
 	}
 
-	private function get_prop_types_to_dynamic_mapping() {
-		return [
-			Image_Prop_Type::get_key() => [
-				Dynamic_Tags_Module::IMAGE_CATEGORY,
-			],
+	/**
+	 * @param array<string, Prop_Type> $schema
+	 *
+	 * @return array<string, Prop_Type>
+	 */
+	private function add_dynamic_prop_type( array $schema ): array {
+		foreach ( $schema as $prop ) {
+			if ( ! ( $prop instanceof Prop_Type ) ) {
+				continue;
+			}
 
-			Number_Prop_Type::get_key() => [
-				Dynamic_Tags_Module::NUMBER_CATEGORY,
-			],
+			if ( false === $prop->get_meta( 'dynamic' ) ) {
+				continue;
+			}
 
-			String_Prop_Type::get_key() => [
-				Dynamic_Tags_Module::TEXT_CATEGORY,
-			],
-		];
+			if ( $prop instanceof Number_Prop_Type ) {
+				$prop->additional_type(
+					Dynamic_Prop_Type::make()->categories( [ Dynamic_Tags_Module::NUMBER_CATEGORY ] )
+				);
+
+				continue;
+			}
+
+			if ( $prop instanceof Image_Prop_Type ) {
+				$prop->additional_type(
+					Dynamic_Prop_Type::make()->categories( [ Dynamic_Tags_Module::IMAGE_CATEGORY ] )
+				);
+
+				continue;
+			}
+
+			if ( $prop instanceof String_Prop_Type && empty( $prop->get_enum() ) ) {
+				$prop->additional_type(
+					Dynamic_Prop_Type::make()->categories( [ Dynamic_Tags_Module::TEXT_CATEGORY ] )
+				);
+
+				continue;
+			}
```

Comment:
Run based on the map?

Reply:
that's what I did at first, but then I started having more conditions (like the enum), and adding an exception for the string using the map was ugly

maybe we can do something like:
```PHP
$map = [
	[
		'condition' => fn( Prop_Type $type ) => $type instanceof Image_Prop_Type
		'categories' => [ IMAGE ],
	],
	[
		'condition' => fn( Prop_Type $type ) => $type instanceof String_Prop_Type && empty( $prop->get_enum() )
		'categories' => [ TEXT ],
	]
]
```

WDYT?

Reply:
Sound good to me.